_aicspylibczi/TypedImage.h

```diff
@@ -123,7 +129,7 @@ namespace pylibczi {
   TypedImage<T>::channelPtrs(int channel_)
   {
       std::pair<T*, T*> ans;
-      size_t planeSize = std::accumulate(m_shape.rbegin(), --(m_shape.rend()), 1, std::multiplies<size_t>());
+      size_t planeSize = std::accumulate(m_shape.rbegin(), --(m_shape.rend()), 1, std::multiplies<size_t>() );
```

Comment:
is decrementing m_shape.rend() always safe?

Reply:
m_shape is a vector so since the memory is contiguous it should be fine. I'm just saying skip the first element in the vector. Wolud you be happier with 
`std::accumulate(++m_shape.begin(), m_shape.end(), 1, std::multiplies<size_t>() ); `

As I understand it --rend() => ++begin() .
Am I following you?

Reply:
can it ever be empty? what happens if you ++ the begin iterator on an empty collection?

Reply:
I started digging into it because of your question and it turns out this function is no longer ever used. I'm going to nuke it since it's not used. The BGR changes made it obsolete.